e2e/connect.test.tsx

```diff
@@ -5,49 +5,86 @@
 import '@ethersproject/providers'
 import 'jest-environment-hardhat'
 
-import { act, render, RenderResult, waitFor } from '@testing-library/react'
+import { render, RenderResult, waitFor } from '@testing-library/react'
 import { tokens } from '@uniswap/default-token-list'
 
 import { SwapWidget } from '../src'
 
 describe('connect', () => {
   let component: RenderResult
-  let wallet: HTMLElement
+  let account: HTMLElement
+  let connectWallet: HTMLElement
   let toolbar: HTMLElement
+  let tokenSelect: HTMLElement
 
   beforeEach(async () => {
     component = render(<SwapWidget tokenList={tokens} />)
-    wallet = await component.findByTestId('wallet')
+    connectWallet = await component.findByTestId('connect-wallet')
     toolbar = await component.findByTestId('toolbar')
+    tokenSelect = (await component.findAllByTestId('token-select'))[0]
   })
 
-  it('prompts for wallet connection in the Toolbar', async () => {
-    expect(wallet.hidden).toBeTruthy()
-    expect(toolbar.textContent).toBe('Connect wallet to swap')
+  describe('with no params, using fallback JSON RPC URL', () => {
+    it('prompts for wallet connection in the Wallet', async () => {
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(connectWallet.textContent).toBe('Connect wallet to swap')
+    })
+    it('prompts for wallet connection in the Toolbar', async () => {
+      expect(toolbar.textContent).toBe('Connecting…')
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(toolbar.textContent).toBe('Connect wallet to swap')
+    })
+    it('expects widget not to be disabled', async () => {
+      expect(tokenSelect).toHaveProperty('disabled', true)
+      await waitFor(() => expect(toolbar.textContent).not.toBe('Connecting…'))
+      expect(tokenSelect).toHaveProperty('disabled', false)
+    })
   })
 
-  describe('with jsonRpcUrlMap', () => {
+  describe('with valid integrator jsonRpcUrlMap', () => {
     it('prompts for wallet connection in the Wallet', async () => {
```

Comment:
These first two tests can be combined (especially as this is an e2e test, and not unit tests).